shell/platform/darwin/ios/ios_context.mm

```diff
@@ -25,6 +26,8 @@
     std::shared_ptr<const fml::SyncSwitch> is_gpu_disabled_sync_switch) {
   switch (api) {
     case IOSRenderingAPI::kSoftware:
+      FML_CHECK(backend != IOSRenderingBackend::kImpeller)
+          << "Software rendering is incompatible with Impeller.";
```

Comment:
nit, maybe something like: "Automatically selected software rendering due to X, but Impeller was enabled.. Impeller does not support software rendering. Either fix X or disable Impeller"

Reply:
Done